src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -467,25 +495,85 @@ private PSObject[] InvokePowerShell(string command, IDictionary<string, object>
             return await task;
         }
 
-        private bool VerifyPSScriptAnalyzerAvailable()
+        /// <summary>
+        /// Create a new runspace pool around a PSScriptAnalyzer module for asynchronous script analysis tasks.
+        /// This uses the PowerShell module API to load the first PSScriptAnalyzer module on the module path with
+        /// with the required minimum version and then creates a runspace pool with that module loaded in the initial
+        /// session state.
+        /// </summary>
+        /// <returns>A runspace pool with PSScriptAnalyzer loaded for running script analysis tasks.</returns>
+        private static RunspacePool CreatePssaRunspacePool(out PSModuleInfo pssaModuleInfo)
         {
             using (var ps = System.Management.Automation.PowerShell.Create())
             {
+                // Run `Get-Module -ListAvailable -Name "PSScriptAnalyzer"`
                 ps.AddCommand("Get-Module")
                     .AddParameter("ListAvailable")
                     .AddParameter("Name", PSSA_MODULE_NAME);
 
                 try
                 {
-                    return ps.Invoke()?.Any() ?? false;
+                    // Get the latest version of PSScriptAnalyzer we can find
+                    pssaModuleInfo = ps.Invoke()?
+                        .Select(psObj => psObj.BaseObject)
+                        .OfType<PSModuleInfo>()
+                        .OrderBy(moduleInfo => moduleInfo.Version)
+                        .FirstOrDefault();
                 }
-                catch (Exception)
+                catch (Exception e)
                 {
-                    return false;
+                    throw new Exception("Unable to find PSScriptAnalyzer module", e);
                 }
+
+                if (pssaModuleInfo == null)
+                {
+                    throw new Exception("Unable to find PSScriptAnalyzer module");
+                }
+
+                // Create a base session state with PSScriptAnalyzer loaded
+                InitialSessionState sessionState = InitialSessionState.CreateDefault2();
+                sessionState.ImportPSModule(new [] { pssaModuleInfo.ModuleBase });
+
+                // RunspacePool takes care of queuing commands for us so we do not
+                // need to worry about executing concurrent commands
+                return RunspaceFactory.CreateRunspacePool(sessionState);
             }
         }
 
         #endregion //private methods
+
+        #region IDisposable Support
+
+        private bool disposedValue = false; // To detect redundant calls
```

Comment:
I wonder if there's a way to set this refactor to prefix this field with `_`.  I always forget to change this

Reply:
Me too!